### PR TITLE
@feature/example 6 content to web

### PR DESCRIPTION
**Pull request by rohit-lakhanpal**

### CODE REVIEW

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  // Retrieve params into a variable
+  const [params, setParams] = useState(useParams());
+
+  // Set Params
+  useEffect(() => {
+    setPanvivaDocument({
+      ...panvivaDocument,
+      id: params.id,
+    });
+  }, [params]);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js]: Add `panvivaDocument` to the dependancy array after `params.`

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  // Fetch document content
+  useEffect(() => {
+    // Set document title
+    //document.title = "Document id ...";
+
+    // todo: check if id is null ??
+    if (panvivaDocument && panvivaDocument.id) {
+      getDocument(panvivaDocument.id);
+    }
+  }, [panvivaDocument.id]);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js]: Just adding `panvivaDocument` to the dependancy array is enough.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+  // Fetch document content
+  useEffect(() => {
+    // Set document title
+    document.title = panvivaDocument.loading
+      ? "Panviva - Loading ..."
+      : `Panviva - ${panvivaDocument?.document?.name} [#${panvivaDocument?.document?.id}]`;
+  }, [panvivaDocument.document]);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js]: Add `panvivaDocument` and `document` to the dependancy array.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  // Todo: Cleanup code
+  const getHtml = (body) => {
+    var htmlObject = document.createElement("div");
+    htmlObject.innerHTML = body;
+    for (
+      let element = 0;
+      element < htmlObject?.getElementsByTagName("img").length;
+      element++
+    ) {
+      debugger;
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js]: Leftover `debugger` statement.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      let imageId = htmlObject
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/Document.js]: Fix the indentation here from line 74 to line 79

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSectionAndSubSectionBody.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+export const DocumentSectionAndSubSectionBody = (props) => {
+  const fixImages = (htmlObject) => {
+    let panvivaImages = htmlObject?.getElementsByTagName("img");
+    for (let idx = 0; idx < panvivaImages.length; idx++) {
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSectionAndSubSectionBody.js]: use [Array.prototype.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) since you are not using the index for anything other that retrieving the object from the array.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSectionAndSubSectionBody.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      let imageId = panvivaImages[idx]?.getAttribute("data-image-id");
+      panvivaImages[idx].setAttribute("src", `api/panviva/image/${imageId}`);
+    }
+    return htmlObject;
+  };
+
+  const fixTooltips = (htmlObject) => {
+    let tooltipClassName = "pv-tooltip-link";
+    let tooltipElements = htmlObject?.getElementsByClassName(tooltipClassName);
+    for (let idx = 0; idx < tooltipElements.length; idx++) {
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSectionAndSubSectionBody.js]: use [Array.prototype.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) since you are not using the index for anything other that retrieving the object from the array.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+import { SearchBar } from "./SearchBar";
+import { SearchResults } from "./SearchResults";
+
+export const Search = () => {
+  const history = useHistory();
+  const [params] = useState(useParams());
+  const [title, setTitle] = useState("");
+  const [pendingQuery, setPendingQuery] = useState("");
+  const [searchQuery, setSearchQuery] = useState(
+    params && params.query !== "" ? params.query : "*"
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js:12]: Can shorten this to `params?.query || "*"`

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  );
+  const [searchResults, setSearchResults] = useState({});
+  const [isLoading, setLoading] = useState(true);
+
+  // update the document title
+  useEffect(() => {
+    document.title = title;
+  }, [title]);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js:20]: dependancy should be `document`.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Error.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+export const Error = () => {
+  const [params, setParams] = useState(useParams());
+  const [message] = useState(
+    params && params.message && params.message !== ""
+      ? params.message
+      : "Undefined error."
+  );
+  const [status] = useState(
+    params && params.status && params.status !== "" ? params.status : "500"
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Error.js]: can shorten to `params?.status || "500"`

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Home.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <div>
+      <div
+        className="d-flex align-content-center flex-wrap"
+        style={{ height: "60vh" }}
+      >
+        <SearchBar
+          handleInputChange={(e) => {
+            setSearchQuery(e.target.value);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Home.js]: Events can sometimes have undefineds or nulls. Just to be safe add `e?.target?.value`

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Layout.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import { Container } from 'reactstrap';
+import { NavMenu } from './NavMenu';
+
+export class Layout extends Component {
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/Layout.js]: Convert to a function component for consistency.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/NavMenu.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+  NavbarBrand,
+  NavbarToggler,
+  NavItem,
+  NavLink,
+} from "reactstrap";
+import { HouseFill } from "react-bootstrap-icons";
+import { Link } from "react-router-dom";
+import "./NavMenu.css";
+
+export class NavMenu extends Component {
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/NavMenu.js]: Convert to a functional component for consistency.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSection.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import { DocumentSubSection } from "./DocumentSubSection";
+import { DocumentSectionAndSubSectionBody } from "./DocumentSectionAndSubSectionBody";
+
+export const DocumentSection = (props) => {
+  return (
+    <div className="m-1">
+      <h2 id={props.section.id}># {props.section.name}</h2>
+
+      {props?.subSecions.map((subSection) => {
+        return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSection.js]: Don't need an explicit return statement.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSubSection.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { DocumentSectionAndSubSectionBody } from "./DocumentSectionAndSubSectionBody";
+
+export const DocumentSubSection = (props) => {
+  return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/document/DocumentSubSection.js:5]: Don't need an explicit return statement.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  useEffect(() => {
+    setLoading(true);
+    if (searchQuery !== "") {
+      searchContent(searchQuery);
+      setTitle(`Panviva - Search results for ${params.query}`);
+    } else {
+      searchContent();
+      setTitle(`Panviva - Search everything!`);
+    }
+  }, [searchQuery]);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js]: Add `params` to the dependancy array.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+        history.push(path);
+      } else {
+        setSearchResults(data);
+      }
+      setLoading(false);
+    } catch (error) {
+      let errorMessage = JSON.stringify(error);
+      let path = `/error/unknown/${errorMessage}`;
+      console.error(errorMessage);
+      console.error(errorMessage);
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js]: Unnecessary log statement.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsListItem.js [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+      return item && item.id ? `/api/panviva/file/${item.id}` : "#";
+    } else if (item.type === "image") {
+      return item && item.id ? `/api/panviva/image/${item.id}` : "#";
+    }
+    return "#";
+  };
+
+  const getFirstDocumentUrl = () => {
+    if (
+      props &&
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsListItem.js]: Can be shortened to `params?.searchResults?.results?.length > 0`

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsList.js [hunk 19]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { SearchResultsListItem } from "./SearchResultsListItem";
+
+export const SearchResultsList = (props) => {
+  return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsList.js:5]: No need for explicit `return`.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsHeading.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { ProgressBar } from "react-bootstrap";
+
+export const SearchResultsHeading = (props) => {
+  return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResultsHeading.js:5]: No need for explicit `return`.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResults.js [hunk 21]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { SearchResultsHeading } from "./SearchResultsHeading";
+import { SearchResultsList } from "./SearchResultsList";
+
+export const SearchResults = (props) => {
+  //isLoading
+  return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResults.js:6]: No need for explicit `return`.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchBar.js [hunk 22]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { Button, InputGroup, FormControl } from "react-bootstrap";
+import { Search } from "react-bootstrap-icons";
+
+export const SearchBar = (props) => {
+  return (
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchBar.js:6]: No need for explicit `return`.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/App.js [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+import { Route } from "react-router";
+import { Layout } from "./components/Layout";
+import { Home } from "./components/Home";
+import { Document } from "./components/document/Document";
+import { Search } from "./components/search/Search";
+import { Error } from "./components/Error";
+
+import "./custom.css";
+
+export default class App extends Component {
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/App.js]: Convert to functional component for consistency.

**Review by r3core — COMMENTED** [review]
Just some overall thoughts.

- I wasn't considering HTML manipulation in this review. I assume you didn't find a good library?
- Usually it's good to use the `.jsx` suffix for filenames for files with React code inside them.
- CSS files + class names inside JSX components is fine. However we should be looking at styled components.
- A lot of error handling code is present within the controller that does nothing but logging. Wouldn't it be better to move to the exception filter?

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js (hunk 9 above)
Comment (rohit-lakhanpal) [reply Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/Search.js:20]: Setting the html document title.

File: Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResults.js [hunk 24]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { SearchResultsHeading } from "./SearchResultsHeading";
+import { SearchResultsList } from "./SearchResultsList";
+
+export const SearchResults = (props) => {
+  //isLoading
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/ClientApp/src/components/search/SearchResults.js]: Remove comment.

File: Samples.Net.Web/Samples.Net.Web/appsettings.Development.json [hunk 25]
```diff
@@ -0,0 +1,9 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
```
Comment (r3core) [inline Samples.Net.Web/Samples.Net.Web/appsettings.Development.json:7]: Add the Panviva settings here as well.

**Review by r3core — APPROVED** [review]
